commands/helpers.go

```diff
@@ -0,0 +1,198 @@
+package commands
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/gruntwork-io/cloud-nuke/config"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/cloud-nuke/telemetry"
+	"github.com/gruntwork-io/cloud-nuke/ui"
+	"github.com/gruntwork-io/go-commons/errors"
+	commonTelemetry "github.com/gruntwork-io/go-commons/telemetry"
+	"github.com/urfave/cli/v2"
+)
+
+// loadConfigFile loads and parses a config file from the given path
+func loadConfigFile(configFilePath string) (config.Config, error) {
+	if configFilePath == "" {
+		return config.Config{}, nil
+	}
+
+	telemetry.TrackEvent(commonTelemetry.EventContext{
+		EventName: "Reading config file",
+	}, map[string]interface{}{})
+
+	configObjPtr, err := config.GetConfig(configFilePath)
+	if err != nil {
+		telemetry.TrackEvent(commonTelemetry.EventContext{
+			EventName: "Error reading config file",
+		}, map[string]interface{}{})
+		return config.Config{}, ConfigFileReadError{FilePath: configFilePath, Underlying: err}
+	}
+
+	return *configObjPtr, nil
+}
+
+// parseAndApplyTimeFilters parses time filter flags and applies them to the config
+func parseAndApplyTimeFilters(c *cli.Context, configObj *config.Config) error {
+	excludeAfter, err := parseDurationParam(FlagOlderThan, c.String(FlagOlderThan))
+	if err != nil {
+		return errors.WithStackTrace(err)
+	}
+
+	includeAfter, err := parseDurationParam(FlagNewerThan, c.String(FlagNewerThan))
+	if err != nil {
+		return errors.WithStackTrace(err)
+	}
+
+	// Apply time filters to config
+	configObj.ApplyTimeFilters(excludeAfter, includeAfter)
+
+	return nil
+}
+
+// parseAndApplyTimeout parses the timeout flag and applies it to the config
+func parseAndApplyTimeout(c *cli.Context, configObj *config.Config) error {
+	timeout, err := parseTimeoutDurationParam(FlagTimeout, c.String(FlagTimeout))
+	if err != nil {
+		return errors.WithStackTrace(err)
+	}
+
+	if timeout != nil {
+		configObj.AddTimeout(timeout)
+	}
+
+	return nil
+}
+
+// confirmNuke handles the nuke confirmation prompt and countdown
+// Returns true if the nuke should proceed, false otherwise
+func confirmNuke(c *cli.Context, hasResources bool) (bool, error) {
```

Comment:
Context state is ignored; I suspect it should be checked if context is done

Reply:
ditto.